### PR TITLE
feat: parse literal mulitline yaml strings

### PR DESCRIPTION
**Pull request by kormide**
fyi: @arobinson

Addresses https://github.com/aspect-build/rules_js/issues/282.

### CODE REVIEW

File: npm/private/test/yaml_tests.bzl [hunk 1]
```diff
@@ -126,6 +126,81 @@ def _parse_basic_test_impl(ctx):
 }
 """))
 
+    # Literal multiline strings (|), strip (-) and keep (+)
+    asserts.equals(env, {"foo":"bar\n" }, parse("""\
+foo: |
+    bar
```
Comment (gregmagolan) [inline npm/private/test/yaml_tests.bzl:132]: would a trailing spaces after `bar` be kept?

Comment (kormide) [reply npm/private/test/yaml_tests.bzl:132]: Yes, they should be. I'll add a test case for that.

Comment (kormide) [reply npm/private/test/yaml_tests.bzl:132]: Added some trailing spaces to the third test case after `baz`. You can't see those spaces in the github diff, but you can see it in the expected value.

Comment (gregmagolan) [reply npm/private/test/yaml_tests.bzl:132]: 🌮

**Pull request merged**